fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/query/WeightEvaluationQuery.java

```diff
@@ -200,7 +205,9 @@ private static DataSource makeInnerQuery(DruidAggregationQuery<?> query, double
                         aggregations,
                         postAggregations,
                         innerQuery.getIntervals(),
-                        ((GroupByQuery) innerQuery).getLimitSpec()
+                        ((GroupByQuery) innerQuery).getLimitSpec() == null ?
+                                null :
+                                ((GroupByQuery) innerQuery).getLimitSpec().withColumns(new LinkedHashSet<>())
```

Comment:
You could also have it take a `Query` instead, and do the type checking and casting inside the method. Either approach is fine.